src/drive/web/modules/navigation/duck/actions.jsx

```diff
@@ -545,14 +545,22 @@ export const openFileWith = (id, filename) => {
   }
 }
 
-export const decryptVaultEncryptionKey = passphrase => {
+export const decryptVaultEncryptionKey = (vault, passphrase) => {
   return async (dispatch, _, { client }) => {
-    // TODO derive secret key + get encrypted vault key + decrypt it
     const salt = client.getStackClient().uri
-    const key = await deriveKey(passphrase, salt)
+    const derivedKey = await deriveKey(passphrase, salt)
+    const wrappedKey = decodeArrayBuffer(vault.key.encrypted_key)
+    const vaultKey = await unwrapAESKey(wrappedKey, derivedKey, {
+      algorithm: 'AES-GCM',
+      keyUsages: vault.key.key_ops
+    }).catch(error => {
+      Alerter.error('encryption.passphrase.incorrect')
+      throw error
+    })
+    // TODO: export/import the key to overcome the firefox AES-KW bug
```

Comment:
OK my bad : it's here. In order for the project to work before a new release of FF, I would have set it directly

Reply:
Well it is planned for the next PR ;)